stcgal/protocols.py

```diff
@@ -238,6 +238,20 @@ def get_iap_delay(self, clock_hz):
 
         return iap_wait
 
+    def delay_safely_written(self, length):
+        """
+        Delay until data has been safely written and sent to device.
+        Some buggy serial drivers don't implement tcdrain/flush correctly.
+        That is, they wait until all data has been written to USB, but they
+        do not wait until the data has actually finished transmission.
+        Add additional delay to work around.
+        """
+
+        bit_time = 1.0 / self.ser.baudrate
+        byte_time = bit_time * 11.0 # start, 8 data bits, stop, parity
+        clock_safety_factor = 1.05 # additional delay in case clock is slow
+        time.sleep(byte_time * length * clock_safety_factor)
```

Comment:
I'd add a command-line option for easier tuning the clock_safety_factor.

Reply:
Are you sure that's needed? If the UART clock is off by more than a few percent, things will break anyway. I'd rather keep it simple. Looking forward to your test results.

Reply:
No, definetely not sure. This needs testing. I can't be sure about anything I've hacked 4 years ago ;)